starter_code/src/movies.js

```diff
@@ -1,19 +1,143 @@
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+function ratesAverage(movies) {
 
+    let container = movies.reduce(function(initial, film) {
+
+        return parseFloat(initial + film.rate)
+
+    }, 0);
+    return +((container/movies.length).toFixed(2))
+}
  
 // Iteration 2: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(movies) {
+    //get all movies that are drama
+    
+    let dramaArray = movies.filter(function(film) {
+        if (film.genre.includes('Drama') == true) {
+            return true;
+        }
+    })
 
+    if (dramaArray.length == 0) {
+        return 0;
+    }
+
+    return ratesAverage(dramaArray)
+}
 
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
+function orderByYear(movies) {
+    let arrayContainer = movies.slice(0);
+
+    arrayContainer.sort(function(a,b) {
+        if (Number(a.year) != Number(b.year)) {
+            return Number(a.year) - Number(b.year);
+        }
+        else if (a.title < b.title) { //don't really understand this one
+            return -1;
+        }
+        else if (a.title > a.title) { //don't really understand this one
+            return 1;
+        }
+        return 0;
+    })
 
+    return(arrayContainer)
+}
 
 // Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct
+function howManyMovies(movies) {
+    //create array of all movies of Spielberg
+    let spielbergArray = movies.filter(function(film) {
+        if (film.director.includes('Steven Spielberg') == true) {
+            return true;
+        }
+    });
 
+    //create array that contains only dramas
+    let dramaSpielbergArray =spielbergArray.filter(function(film) {
+        if (film.genre.includes('Drama') == true) {
```

Comment:
In this case you didn't have to the the == true, because the includes method already returns a boolean. And if a value is true, or truthy, than the if statement runs. No need to add that extra code :)